tests/test_schemas.py

```diff
@@ -172,22 +216,274 @@ def test_sub_schema_graph(self):
         return
 
 class TestDataModelNodes:
-    def test_gather_nodes(self):
-        return
-    def test_gather_all_nodes(self):
-        return
-    def test_get_rel_node_dict_info(self):
-        return
-    def test_get_data_model_properties(self):
-        return
-    def test_get_entry_type(self):
-        return
-    def test_run_rel_functions(self):
-        return
-    def test_generate_node_dict(self):
-        return
-    def test_generate_node(self):
-        return
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    def test_gather_nodes(self, helpers, data_model):
+        # Instantiate Parser
+        data_model_parser = helpers.get_data_model_parser(data_model_name=data_model)
+
+        # Parse Model
+        attr_rel_dictionary = data_model_parser.parse_model()
+
+        # Instantiate DataModelNodes
+        data_model_nodes = generate_data_model_nodes(helpers, data_model_name=data_model)
+
+        attr_info = ('Patient', attr_rel_dictionary['Patient'])
+        nodes = data_model_nodes.gather_nodes(attr_info=attr_info)
+
+        # Make sure there are no repeat nodes
+        assert len(nodes) == len(set(nodes))
+
+        # Make sure the nodes returned conform to expectations (values and order)
+        ## The parsing records display names for relationships for CSV and labels for JSONLD, so the expectations are different between the two.
+        if DATA_MODEL_DICT[data_model]=='CSV':
+            expected_nodes = ['Patient', 'Patient ID', 'Sex', 'Year of Birth', 'Diagnosis', 'Component', 'DataType']
+        elif DATA_MODEL_DICT[data_model] == 'JSONLD':
+            expected_nodes = ['Patient', 'PatientID', 'Sex', 'YearofBirth', 'Diagnosis', 'Component', 'DataType']
+
+        assert nodes == expected_nodes
+
+        # Ensure order is tested.
+        reordered_nodes = nodes.copy()
+        reordered_nodes.remove('Patient')
+        reordered_nodes.append('Patient')
+        assert reordered_nodes != expected_nodes
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    def test_gather_all_nodes(self, helpers, data_model):
```

Comment:
I understand what you are trying to do here. I am thinking since `gather_all_nodes` calls `gather_nodes` function, will it make sense to mock `gather_nodes` and then test if `gather_all_nodes` could return the correct output?